src/components/TranslationDialog/TranslationModal/useTranslationsResults.js

```diff
@@ -0,0 +1,28 @@
+import { useAlert, useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { useRef } from 'react'
+
+export const useTranslationsResults = ({ resource }) => {
+    const translationsQueryRef = useRef({
+        translations: {
+            resource: `${resource}/translations`,
+        },
+    })
+
+    const { data, fetching, refetch } = useDataQuery(
+        translationsQueryRef.current,
+        {
+            onError: error => showError(error),
+        }
+    )
+
+    const { show: showError } = useAlert(error => error.message, {
```

Comment:
I'm not sure how correct this is right now, but in the past I have ran into situations where the error didn't have a `message` property (I believe the error contained a more complex object with an error report 🤷 ). Perhaps we should implement a fallback message?
```suggestion
    const fallbackMsg = i18n.t('Could not load translations')
    const { show: showError } = useAlert(error => error.message || fallbackMsg, {
```

Reply:
Added.